test/helpers/kubectl.go

```diff
@@ -2916,36 +2917,64 @@ func (kub *Kubectl) GetCiliumPodOnNodeWithLabel(namespace string, label string)
 	return kub.GetCiliumPodOnNode(namespace, node)
 }
 
-func (kub *Kubectl) ciliumPreFlightCheck() error {
-	err := kub.ciliumStatusPreFlightCheck()
-	if err != nil {
-		return fmt.Errorf("status is unhealthy: %s", err)
-	}
+func (kub *Kubectl) validateCilium() error {
+	var (
+		wg     sync.WaitGroup
+		errors = make(chan error, 128)
+	)
 
-	err = kub.ciliumControllersPreFlightCheck()
-	if err != nil {
-		return fmt.Errorf("controllers are failing: %s", err)
-	}
+	wg.Add(1)
+	go func() {
+		if err := kub.ciliumStatusPreFlightCheck(); err != nil {
+			errors <- fmt.Errorf("status is unhealthy: %s", err)
+		}
+		wg.Done()
+	}()
+
+	wg.Add(1)
+	go func() {
+		if err := kub.ciliumControllersPreFlightCheck(); err != nil {
+			errors <- fmt.Errorf("controllers are failing: %s", err)
+		}
+		wg.Done()
+	}()
 
 	switch GetCurrentIntegration() {
```

Comment:
This reads as if the switch statement is not really needed any more...

Reply:
I guess we still don't want to run connectivity check while running Flannel integration.

Reply:
I see, I forgot that in Go there is no implicit fall-through, my bad. Perhaps an if statement  instead of a switch would make this more readable?